src/components/Modals/PublishCollectionModal/PublishCollectionModal.tsx

```diff
@@ -1,39 +1,120 @@
 import * as React from 'react'
-import { ModalNavigation, Button } from 'decentraland-ui'
+import { ModalNavigation, Button, Mana, Loader } from 'decentraland-ui'
 import Modal from 'decentraland-dapps/dist/containers/Modal'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 
-import { Props } from './PublishCollectionModal.types'
+import { builder } from 'lib/api/builder'
+import { fromWei } from 'web3x-es/utils'
+import { RARITY_COLOR, ItemRarity } from 'modules/item/types'
+import { Props, State } from './PublishCollectionModal.types'
 import './PublishCollectionModal.css'
 
-export default class PublishCollectionModal extends React.PureComponent<Props> {
-  componentDidMount() {
+export default class PublishCollectionModal extends React.PureComponent<Props, State> {
+  state: State = { step: 1, rarities: [], isFetchingRarities: true }
+
+  async componentDidMount() {
     const { collection, onClose } = this.props
     if (!collection) {
       onClose()
     }
+
+    const rarities = await builder.fetchRarities()
+    this.setState({ rarities, isFetchingRarities: false })
```

Comment:
Should we add this to the "hande error states" issue? (as it'd load forever, right? there's no `trycatch`). Wouldn't fix it here.